lib/std/fs/file.zig

```diff
@@ -271,6 +271,8 @@ pub const File = struct {
     /// The `iovecs` parameter is mutable because this function needs to mutate the fields in
     /// order to handle partial reads from the underlying OS layer.
     pub fn readvAll(self: File, iovecs: []os.iovec) ReadError!void {
+        if (iovecs.len == 0) return;
```

Comment:
I don't this we want this check?
In additional to making debugging hard and potential performance issues (I've encounted code before that checks for empty vectors at every level.... the program was spending more time in these checks then if it just *did* the thing): this may also have an external effect in terms of
  - not updating the `atime` of a file
  - missing out on a permission denied error

Reply:
Generally i agree. In this case it has to otherwise the [0] later in the function will OOB